### ISSUE TITLE
Unable to find devices

### ISSUE THREAD

**Issue by matthew-jones-uk** [balloob/pychromecast#228]
Hi, so I seem to be having some trouble detecting my gen 1 chromecast and my Google home mini. Running pychromecast.get_chromecasts() on both a Windows machine and a Linux one. It's probably a problem with my network but I would be grateful if you could give me any ideas on how to solve it.

Thanks.

**Comment by anthonyzhub** [comment]
Hi,

Would you mind posting your code? Your code should like this:

`# Locate any Google Devices connected to network
chromecasts = pychromecast.get_chromecasts()

# Of all the devices from 'chromecasts' look for your device
cast = next(cc for cc in chromecasts if cc.device.friendly_name == "device name here")`

**Comment by matthew-jones-uk** [comment]
Running chromecasts = pychromecast.get_chromecasts() returns nothing, however, my Android phone is able to see the chromecast.

Also quick question - does this work with a Google Home?

**Comment by matthew-jones-uk** [comment]
Ah... No, I've been using Python 3. Any way to fix the issues?

**Comment by anthonyzhub** [comment]
Ok. You may be right about having network issues. On my computer, I permitted my program to either receive incoming or sending outcoming connections. Check your security settings on one of your computers.

**Comment by matthew-jones-uk** [comment]
All security settings should allow it, and I tried using multiple computers.

**Comment by rseward** [comment]
In the past, I have had to disable the firewall on my Linux
workstation, to stream stuff to my ChromeCasts. I never did figure out
the necessary firewall rules I would need to write to allow
communication to the chromecasts, so I would just disable my Linux
firewall all together when I wanted to cast.


On Thu, 2018-05-03 at 17:54 -0700, anthonyzhub wrote:
> Hi,
> Would you mind posting your code? Your code should like this:
> `# Locate any Google Devices connected to network
> chromecasts = pychromecast.get_chromecasts()
> Of all the devices from 'chromecasts' look for your device
> cast = next(cc for cc in chromecasts if cc.device.friendly_name ==
> "device name here")`
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
-- 
Rob Seward
Bluestone Consulting Group, LLC

web:     http://www.bluestone-consulting.com/
e-mail:  rseward@bluestone-consulting.com
office:   734.726.0313

**Comment by matthew-jones-uk** [comment]
Resolved, a reset of my chromecast and restart of my Home seemed to fix it.

**Comment by ikkysleepy** [comment]
I have similar issues. I am thinking there is something wrong with the mdns / zeroconfig. I can connect to my chromecast but while testing, sending play pause commands and turning on  / off my tv, several times the scan results to nothing for a dozen times... I repeat the command o find any chromecast, but I get no results. I noticed if I pause or play with another device that sometimes it does pickup chromecast. I ran another mdns scan and it was able to find the chromecast even though pychromecast was not. I will enable debugging to see what is going on but something is not right that will cause the scan to not find anything.
